components/org.wso2.carbon.business.rules.core/src/main/java/org/wso2/carbon/business/rules/core/util/TemplateManagerHelper.java

```diff
@@ -69,23 +67,20 @@ private TemplateManagerHelper() {
     /**
      * Converts given JSON File to a JSON object
      *
-     * @param jsonFile
-     * @return
-     * @throws TemplateManagerException
+     * @param jsonFile json file
+     * @return JsonObject
+     * @throws TemplateManagerException exceptions related to business rules
      */
     public static JsonObject fileToJson(File jsonFile) throws TemplateManagerException {
         Gson gson = new GsonBuilder().disableHtmlEscaping().setPrettyPrinting().create();
-        JsonObject jsonObject = null;
-
+        JsonObject jsonObject;
         try {
-            Reader reader = new BufferedReader(new InputStreamReader(new FileInputStream(jsonFile), Charset.forName("UTF-8")));
+         Reader reader = new BufferedReader(new InputStreamReader(new FileInputStream(jsonFile),
```

Comment:
once done awe need to close the reader in a finally block

Reply:
Also, you can initialize the reader inside the try(Reader reader ...){}
Please check the Try-with-resource

Reply:
Fixed by 1abe5bdb86d2f887e1a4b2964d6f0d931ede298f